File: doc/codes/python/examples/api/compute/pattern/set.py
from autd3.geometry import Autd3, Geometry
from autd3.value import Intensity, Phase
from autd3_pattern import add_phase, set_intensity, set_phase

geometry = Geometry([Autd3([0.0, 0.0, 0.0], [1.0, 0.0, 0.0, 0.0])])

intensity = Intensity(0x80)
phase = Phase.PI
phases = geometry.phase_buffer()
intensities = geometry.intensity_buffer()

# ANCHOR: set_intensity
set_intensity(intensity, intensities)
# ANCHOR_END: set_intensity

# ANCHOR: set_phase
set_phase(phase, phases)
# ANCHOR_END: set_phase

# ANCHOR: add_phase
add_phase(phase, phases)
# ANCHOR_END: add_phase

File: doc/codes/rust/examples/api/compute/pattern/set.rs
use autd3_rs::geometry::{Autd3, Geometry};
use autd3_rs::value::{Intensity, Phase};
use autd3_rs_pattern::{add_phase, set_intensity, set_phase};

fn main() {
    let geometry = Geometry::new(vec![Autd3::default()]);

    let intensity = Intensity(0x80);
    let phase = Phase::PI;
    let mut phases = geometry.phase_buffer();
    let mut intensities = geometry.intensity_buffer();

    // ANCHOR: set_intensity
    set_intensity(intensity, &mut intensities);
    // ANCHOR_END: set_intensity

    // ANCHOR: set_phase
    set_phase(phase, &mut phases);
    // ANCHOR_END: set_phase

    // ANCHOR: add_phase
    add_phase(phase, &mut phases);
    // ANCHOR_END: add_phase
}
